runtime/commonMain/src/kotlinx/serialization/Exceptions.kt

```diff
@@ -5,25 +5,70 @@
 package kotlinx.serialization
 
 /**
- * A generic exception indicating the problem during serialization or deserialization process
+ * A generic exception indicating the problem during serialization or deserialization process.
+ * This is a generic exception type that can be thrown during the problem at any stage of the serialization,
+ * including encoding, decoding, serialization, deserialization.
+ * [SerialFormat] implementors should throw the subclass of this exception at any
  */
-public open class SerializationException(message: String, cause: Throwable? = null) : RuntimeException(message, cause)
+public open class SerializationException : IllegalArgumentException {
+    /*
+     * Rationale behind making it IllegalArgumentException:
```

Comment:
This is intended to be just a comment, right?

Reply:
Yes